### PR TITLE
[crystal/es-es] spanish translation for crystal

### PR DESCRIPTION
**Pull request by ElCodigoDeNito**
- [x] I solemnly swear that this is all original content of which I am the original author
- [x] Pull request title is prepended with `[language/lang-code]`
- [x] Pull request touches only one file (or a set of logically related files with similar changes made)
- [x] Content changes are aimed at *intermediate to experienced programmers* (this is a poor format for explaining fundamental programming concepts)
- [x] If you've changed any part of the YAML Frontmatter, make sure it is formatted according to [CONTRIBUTING.md](https://github.com/adambard/learnxinyminutes-docs/blob/master/CONTRIBUTING.markdown)
  - [x] Yes, I have double-checked quotes and field names!

### CODE REVIEW

File: es-es/crystal-es.html.markdown [hunk 1]
```diff
@@ -0,0 +1,542 @@
+---
+idioma: cristal
```
Comment (megatux) [inline es-es/crystal-es.html.markdown]: I don't think it is valid to translate the language name. Check other translations, Python is not "Pitón", Dart is not "dardo", Ruby is not "Rubí"

File: es-es/crystal-es.html.markdown [hunk 2]
```diff
@@ -0,0 +1,542 @@
...
+nombre de archivo: learncrystal-es.cr
+colaboradores:
+    - ["Vitalii Elenhaupt", "http://veelenga.com"]
+    - ["Arnaud Fernandés", "https://github.com/TechMagister/"]
+traductores:
+    - ["Juan Vallejo", "https://github.com/ElCodigoDeNito"]
+lang: es-es
+---
+
+```Cristal
```
Comment (megatux) [inline es-es/crystal-es.html.markdown]: Crystal

File: es-es/crystal-es.html.markdown [hunk 3]
```diff
@@ -0,0 +1,542 @@
...
+0b1101 # => 13: Int32
+
+# Número en base octal
+0o123 # => 83: Int32
+
+# Números hexadecimales
+0xFE012D # => 16646445: Int32
+0xfe012d # => 16646445: Int32
+
+# Números de comas
```
Comment (megatux) [inline es-es/crystal-es.html.markdown]: # Números de coma flotante ? https://es.wikipedia.org/wiki/Coma_flotante

File: es-es/crystal-es.html.markdown [hunk 4]
```diff
@@ -0,0 +1,542 @@
...
+# El acceso es a través de una clave
+hash = {"color" => "verde", "número" => 5}
+hash ["color"] # => "verde"
+hash ["no_such_key"] # => Falta la clave hash: "no_such_key" (KeyError)
+desmenuzar [ "no_such_key"]? # => nulo
+
+# Compruebe si hay una clave
+hash.has_key? "color" # => verdadero
+
+# Notación especial para teclas si son símbolos
```
Comment (megatux) [inline es-es/crystal-es.html.markdown]: If we follow with other places translations,  "teclas" should be "claves" , I think.

File: es-es/crystal-es.html.markdown [hunk 5]
```diff
@@ -0,0 +1,542 @@
...
+# existe mientras el tipo defina un método .new y # [] =
+clase MyType
+  def [] = (clave, valor)
+    pone "hacer cosas"
+  final
+final
+
+MyType {"foo" => "bar"}
+
+# ¿Qué es equivalente a:
```
Comment (megatux) [inline es-es/crystal-es.html.markdown]: Why the "¿"

File: es-es/crystal-es.html.markdown [hunk 6]
```diff
@@ -0,0 +1,542 @@
...
+hasta índice> 3
+  pone "Índice: # {index}"
+  índice + = 1
+final
+# Índice: 0
+# Índice: 1
+# Índice: 2
+# Índice: 3
+
+# Pero es mejor usar #cada uno
```
Comment (megatux) [inline es-es/crystal-es.html.markdown]: #cada  --->  #each   , the keyword

File: es-es/crystal-es.html.markdown [hunk 7]
```diff
@@ -0,0 +1,542 @@
...
+suma 3, 4 # => 7
+
+suma total (3, 4), 5 # => 12
+
+# rendimiento
+# Todos los métodos tienen un parámetro opcional y un tipo de bloque implícito
+# se puede llamar con la palabra clave 'rendimiento'
+def surround
+  pone '{'
+  rendimiento
```
Comment (megatux) [inline es-es/crystal-es.html.markdown]: "yield"

File: es-es/crystal-es.html.markdown [hunk 8]
```diff
@@ -0,0 +1,542 @@
...
+(1..3) .cada do | index |
+  pone "Índice: # {index}"
+final
+# Índice: 1
+# Índice: 2
+# Índice: 3
+
+# El tipo de una variable depende del tipo de expresión
+# en la declaración del if
+si a <3
```
Comment (megatux) [inline es-es/crystal-es.html.markdown]: Why are we translation the language keywords ?

File: es-es/crystal-es.html.markdown [hunk 9]
```diff
@@ -0,0 +1,542 @@
...
+# operación destructiva. Algunos métodos tienen una versión "!" para hacer
+# cambios y un no "!" para devolver una nueva versión
+company_name = "Dunder Mifflin"
+company_name.gsub "Dunder", "Donald" # => "Donald Mifflin"
+company_name # => "Dunder Mifflin"
+company_name.gsub! "Dunder", "Donald"
+nombre_empresa # => "Donald Mifflin"
+
+
+# Las clases se definen con la palabra clave "clase"
```
Comment (megatux) [inline es-es/crystal-es.html.markdown]: class

**Comment by divayprakash** [comment]
@ElCodigoDeNito Please review the changes as suggested by @megatux

**Comment by verhovsky** [comment]
Closing this, I wanted to fix it but there's so many translated keywords that need to be un-translated

**Pull request closed without merging**